source/DmapSupport.cpp

```diff
@@ -220,10 +227,21 @@ HRESULT GetControllerBaseAddress(PWCHAR deviceName, HANDLE & handle, PVOID & bas
 
     auto asyncAction = ThreadPool::RunAsync(workItem);
 
-    findCompleted->wait();
-
-    baseAddress = (PVOID)controllerAddress;
+    DWORD dwError = WaitForSingleObjectEx(findCompleted, WAIT_TIME_MILLIS, FALSE);
+    if (dwError == WAIT_TIMEOUT)
+    {
+        hr = E_ABORT;
```

Comment:
I'm not sure this is an appropriate HRESULT to return on a timeout.  WAIT_TIMEOUT appears to be a Win32 error code--perhaps HRESULT_FROM_WIN32(WAIT_TIMEOUT) would handle this case...?